src/components/network/map-equipments.js

```diff
@@ -0,0 +1,281 @@
+
+        Promise.all([updatedEquipments])
+            .then((values) => {
+                this.updateSubstations(
+                    this.checkAndGetValues(values[0].substations),
+                    isFullReload
+                );
+                this.updateLines(
+                    this.checkAndGetValues(values[0].lines),
+                    isFullReload
+                );
+                handleUpdatedLines(values[0].lines);
+            })
+            .catch((error) => {
+                console.error(error.message);
+                if (this.errHandler) {
+                    this.errHandler(
+                        this.intlRef.current.formatMessage({
+                            id: 'MapEquipmentsLoadError',
+                        })
+                    );
+                }
+            });
+    }
+
+    completeSubstationsInfos() {
+        const nominalVoltagesSet = new Set();
+        this.substationsById = new Map();
+        this.voltageLevelsById = new Map();
+        this.substations.forEach((substation) => {
+            // sort voltage levels inside substations by nominal voltage
+            substation.voltageLevels = substation.voltageLevels.sort(
+                (voltageLevel1, voltageLevel2) =>
+                    voltageLevel1.nominalVoltage - voltageLevel2.nominalVoltage
+            );
+
+            this.substationsById.set(substation.id, substation);
+
+            substation.voltageLevels.forEach((voltageLevel, index) => {
+                // add substation id and name
+                voltageLevel.substationId = substation.id;
+                voltageLevel.substationName = substation.name;
+
+                // add the current item into the VL map by id
+                this.voltageLevelsById.set(voltageLevel.id, voltageLevel);
+
+                nominalVoltagesSet.add(voltageLevel.nominalVoltage);
+            });
+        });
+
+        this.voltageLevels = Array.from(this.voltageLevelsById.values());
+
+        this.nominalVoltages = Array.from(nominalVoltagesSet).sort(
+            (a, b) => b - a
+        );
+    }
+
+    updateEquipments(currentEquipments, newEquipements, equipmentType) {
+        // replace current modified equipments
+        currentEquipments.forEach((equipment1, index) => {
+            const found = newEquipements.filter(
+                (equipment2) => equipment2.id === equipment1.id
+            );
+            currentEquipments[index] = found.length > 0 ? found[0] : equipment1;
+        });
+
+        // add newly created equipments
+        const eqptsToAdd = newEquipements.filter(
+            (eqpt) =>
+                !currentEquipments.some((otherEqpt) => otherEqpt.id === eqpt.id)
+        );
+        if (eqptsToAdd.length === 0) return currentEquipments;
+        return [...currentEquipments, ...eqptsToAdd];
+    }
+
+    updateSubstations(substations, fullReload) {
+        if (fullReload) {
+            this.substations = [];
```

Comment:
To my knowledge and after testing it, emptying the collection doesn't trigger a map reload right away. 
Through some tests I checked updateState methods inside the layers components are triggered only once. 
This addition was done in order to handle equipments deletions in manual mode since we can't rely on notifications to remove equipments in this mode.